### PR TITLE
Relationships

### PR DESCRIPTION
**Pull request by Sajeelzafar**
### In this pull request, the one-to-many relationship between classroom and student, and many to many relationships between rentals, books, and person is created using Ruby.

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+  def initialize(date, book, person)
+    @date = date
+
+    @book = book
+    book.rentals << self
+
+    @person = person
+    person.rentals << self
+  end
```
Comment (thecodechaser) [inline book.rb]: - [x] Please move the `Rental` class to a separate file `rental.rb` and make sure book, and person variables are also writable so move them to `attr_accessor`.

File: person.rb [hunk 2]
```diff
@@ -48,6 +50,21 @@ def correct_name
...
+  def initialize(label)
+    @label = label
+    @students = []
+  end
+
+  def new_student(student)
+    @students.push(student)
+    student.classroom = self
+  end
+end
```
Comment (thecodechaser) [inline person.rb]: - [x] Please move the `Classroom` class to a separate file `classroom.rb` and move students and label variables in the same `attr_accessor`, you should not duplicate `attr_accessor`.

File: book.rb [hunk 3]
```diff
@@ -0,0 +1,25 @@
+class Book
+  attr_reader :rentals
+  attr_accessor :title, :author
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+end
```
Comment (thecodechaser) [inline book.rb]: - [x] Please make sure in `Book` class a method for adding rentals should be present and make sure the rentals variable is also writable so move it to `attr_accessor`.
  - You can consider the below code

```ruby
 def add_rental(person, date)
    Rental.new(date, self, person)
  end
```

File: person.rb [hunk 4]
```diff
@@ -5,16 +5,18 @@ def correct_name
 end
 
 class Person < Nameable
+  attr_reader :rentals, :id, :parent_permission
+
   def initialize(age, name = 'Unknown', parent_permission: true)
     @id = Random.rand(1..1000)
     @name = name
     @age = age
     @parent_permission = parent_permission
```
Comment (thecodechaser) [inline person.rb:14]: - [x] Please make sure in the `Person` class a method for adding rentals should be present and make sure the rentals variable is also writable so move it to `attr_accessor`.
  - You can consider the below code

```ruby
def add_rental(book, date)
    Rental.new(date, book, self)
  end
```
```

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @Sajeelzafar👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
- No linter errors. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: book.rb (hunk 1 above)
Comment (Sajeelzafar) [reply book.rb]: Thank you for your feedback. I have rectified the highlighted mistake.

File: book.rb (hunk 3 above)
Comment (Sajeelzafar) [reply book.rb]: Thank you for your feedback. I have rectified the highlighted mistake.

File: person.rb (hunk 4 above)
Comment (Sajeelzafar) [reply person.rb:14]: Thank you for your feedback. I have rectified the highlighted mistake.

File: person.rb (hunk 2 above)
Comment (Sajeelzafar) [reply person.rb]: Thank you for your feedback. I have rectified the highlighted mistake.

**Review by EricMbouwe — APPROVED** [review]
Hi @Sajeelzafar, 

Your project is complete! ⚡🔥  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 🎉🔥
- Made requested changes ✔
- No linter errors ✔

## Optional suggestions ☂

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**